### PR TITLE
IN-2226: filter node metrics

### PR DESCRIPTION
**Pull request by blockloop**

### CODE REVIEW

File: pkg/collector/node.go [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+	"fmt"
+	"strings"
+
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/node_exporter/collector"
 )
 
+var whitelist = map[string]bool{
+	"node_network_receive_bytes_total":  true,
```
Comment (bsnyder788) [inline pkg/collector/node.go]: Seems odd to use a map for a whitelist. if you are going to use a map with a bool value, why not actually check the value of true or false when evaluating the whitelist below?

Comment (Hoenn) [reply pkg/collector/node.go]: :point_up:

Comment (blockloop) [reply pkg/collector/node.go]: The map is only being used as a Set. The value is useless.

File: pkg/collector/node.go [hunk 2]
```diff
@@ -38,12 +57,27 @@ func (n *NodeCollector) Collectors() map[string]collector.Collector {
...
+	go func() {
+		defer close(tee)
+		n.collectFunc(tee)
+	}()
+	for m := range tee {
+		// Desc doesn't allow access to underlying fields like fqName. The String() output contains
+		// Desc{fqName: "node_network_transmit_bytes_total", help: "Network device statistic transmit_bytes.", constLabels: {}, variableLabels: [device]}
+		// this is ugly but currently all we can do
+		d := strings.ToLower(m.Desc().String())
+		for s := range whitelist {
```
Comment (thorfour) [inline pkg/collector/node.go]: parse out the fqname and do a map lookup instead.

Alternatively you can do something like 
```
metric := &dto.Metric{}
m.Write(metric)
```
and pase the metric labels for `__name__` to determine the metric name.

File: pkg/collector/node.go (hunk 1 above)
Comment (bsnyder788) [reply pkg/collector/node.go]: I get that, but you are never doing a lookup so it doesn't really make sense to use a map vs just an array or struct imo

File: pkg/collector/node.go (hunk 2 above)
Comment (blockloop) [reply pkg/collector/node.go]: If I parse it out I'll have to use regex which is easily tenfold slower than just a string search. I'll try with creating a dto.Metric

File: pkg/collector/node.go (hunk 1 above)
Comment (thorfour) [reply pkg/collector/node.go]: :point_up:

File: pkg/collector/node.go (hunk 2 above)
Comment (blockloop) [reply pkg/collector/node.go]: Update: that introduces quite a bit more code by having to check for errors on the Write, looping on label values, etc. What I have feels like the simplest solution :slightly_frowning_face:

Comment (thorfour) [reply pkg/collector/node.go]: it might be slower, but it's probably a more "correct" way to do things.

Comment (thorfour) [reply pkg/collector/node.go]: I mean, simple and incorrect vs. complex and correct.

File: pkg/collector/node.go (hunk 1 above)
Comment (blockloop) [reply pkg/collector/node.go]: Yep. I realize that I changed it from a lookup to a loop now. Pushed a change.

File: pkg/collector/node.go (hunk 2 above)
Comment (blockloop) [reply pkg/collector/node.go]: After attempts to use the m.Write and other methods I haven't found a way to access the fqName.

**Pull request merged**